solr/core/src/java/org/apache/solr/response/transform/ValueSourceAugmenter.java

```diff
@@ -38,14 +40,23 @@
  * @since solr 4.0
  */
 public class ValueSourceAugmenter extends DocTransformer {
+
   public final String name;
   public final QParser qparser;
   public final ValueSource valueSource;
+  private static final Object NULL_SENTINEL = new Object();
+  private static final int MAX_PREFETCH_DEFAULT = 1000;
+  private final int maxPrefetchSize;
 
   public ValueSourceAugmenter(String name, QParser qparser, ValueSource valueSource) {
     this.name = name;
     this.qparser = qparser;
     this.valueSource = valueSource;
+    var localParams = qparser.getLocalParams();
+    maxPrefetchSize =
+        localParams != null
+            ? localParams.getInt("preFetchDocs", MAX_PREFETCH_DEFAULT)
+            : MAX_PREFETCH_DEFAULT;
```

Comment:
oh I recall qparser.getParam(str) now, so you don't have to do the null check and it'll look in the top level request.  It doesn't have the integer parse but you could use `"1000"` (as a literal; no constant needed!)